src/execution/sema/sema_builtin.cpp

```diff
@@ -2395,6 +2405,88 @@ void Sema::CheckBuiltinParamCall(ast::CallExpr *call, ast::Builtin builtin) {
 void Sema::CheckBuiltinStringCall(ast::CallExpr *call, ast::Builtin builtin) {
   ast::BuiltinType::Kind sql_type;
   switch (builtin) {
+    case ast::Builtin::Chr: {
+      // check to make sure this function has two arguments
+      if (!CheckArgCount(call, 2)) {
+        return;
+      }
+
+      // checking to see if the first argument is an execution context
+      auto exec_ctx_kind = ast::BuiltinType::ExecutionContext;
+      if (!IsPointerToSpecificBuiltin(call->Arguments()[0]->GetType(), exec_ctx_kind)) {
+        ReportIncorrectCallArg(call, 0, GetBuiltinType(exec_ctx_kind)->PointerTo());
+        return;
+      }
+
+      // checking to see if the second argument is a number
+      auto *resolved_type = Resolve(call->Arguments()[1]);
```

Comment:
By this point, all argument types have been resolved so you can skip this step and directly use `call->Arguments()[1]->GetType()`. This'll save a costly sub-tree traversal.

Reply:
Fixed